include/UNIXSocketRequest.hpp

```diff
@@ -40,7 +40,8 @@ class UNIXSocketRequest final
         const URL& url,
         std::function<void(const std::string&)> onSuccess,
         std::function<void(const std::string&)> onError = [](auto) {},
-        const std::string& fileName = "");
+        const std::string& fileName = "",
+        const unsigned int retryAttempts = 1);
```

Comment:
no retry as default value.

```suggestion
        const unsigned int retryAttempts = 0);
```